docs/src/ref/distributions.md

```diff
@@ -19,3 +17,207 @@ exponential
 laplace
 ```
 
+## Creating New Distributions via the Distributions DSL
+
+In addition to using the above built-in distributions, and to defining subtypes
+of `Gen.Distribution` in plain Julia, you may combine existing distributions to
+create new ones using the `@dist` DSL.  The syntax for this DSL is
+
+```julia
+@dist name(arg1, arg2, ..., argN) = body
+```
+or
+```julia
+@dist function name(arg1, arg2, ..., argN)
+    body
+end
+```
+
+Here `body` is ordinary Julia code, with the constraint that `body` must
+contain exactly one random choice.  The value of the `@dist` expression is then
+a `Gen.distribution` object called `name`, parameterized by `arg1, ..., argN`,
+representing the distribution over _return values_ of `body`.
+
+This DSL is designed to address the issue that sometimes, values stored in the
+trace do not correspond to the most natural physical elements of the model
+state space, making inference programming and querying more taxing than
+necessary. For example, suppose we have a model of classes at a school, where
+the number of students is random, with mean 10, but always at least 3. Rather
+than writing the model as
+
+```julia
+@gen function class_model()
+   n_students = @trace(poisson(7), :n_students_minus_3) + 3
+   ...
+end
+```
+
+and thinking about the random variable `:n_students_minus_3`, you can use the
+`@dist` DSL to instead write
+
+```julia
+@dist student_distr(mean, min) = poisson(mean-min) + min
+
+@gen function class_model()
+   n_students = @trace(student_distr(10, 3), :n_students)
+   ...
+end
+```
+
+and think about the more natural random variable `:n_students`.  This leads to
+more natural inference programs, which can constrain and propose directly to
+the `:n_students` trace address.
+
+### Permitted means of combination
+
+Of course, it is not possible for `@dist` to work on any arbitrary `body`.  We
```

Comment:
Maybe it is not obvious to readers of the docs -- so perhaps we can remove "Of course"?

Reply:
Done